plugin-flex-ts-template-v2/src/feature-library/custom-transfer-directory/custom-components/QueueItem.tsx

```diff
@@ -0,0 +1,115 @@
+
+  const agents_available = !available || !eligible ? 'Unknown' : `${available}/${eligible}`;
+  // eslint-disable-next-line no-eq-null, eqeqeq
+  const tasks_in_queue = tasks != null && tasks >= 0 ? `${tasks}` : 'Unknown';
+
+  const queue_tooltip = showRealTimeQueueData()
+    ? `Agents: ${agents_available}, Tasks in queue: ${tasks_in_queue}`
+    : `${queue.name}`;
+
+  return (
+    <Stack
+      element="TRANSFER_DIR_QUEUE_HORIZONTAL_ROW_CONTAINER"
+      orientation="horizontal"
+      spacing="space40"
+      key={`queue-item-container-${queue.sid}`}
+    >
+      <ProductContactCenterTeamsIcon
+        key={`queue-item-${queue.sid}`}
+        element="TRANSFER_DIR_COMMON_ROW_ICON"
+        decorative={false}
+        title={queue.name}
+      />
+
+      <Tooltip key={`queue-item-label-tooltip-${queue.sid}`} element="TRANSFER_DIR_COMMON_TOOLTIP" text={queue_tooltip}>
+        <Text
+          key={`queue-item-label-${queue.sid}`}
+          element="TRANSFER_DIR_COMMON_ROW_NAME"
+          as="div"
+          title={queue.name}
+          className="Twilio"
+        >
+          {queue.name}
+        </Text>
+      </Tooltip>
+
+      <ButtonGroup key={`queue-item-buttongroup-${queue.sid}`} element="TRANSFER_DIR_COMMON_ROW_BUTTONGROUP" attached>
+        <Tooltip
+          key={`queue-item-buttons-warm-transfer-tooltip-${queue.sid}`}
+          element="TRANSFER_DIR_COMMON_TOOLTIP"
+          text="Warm Transfer"
+        >
+          <Button
+            key={`queue-item-warm-transfer-button-${queue.sid}`}
+            variant="secondary_icon"
+            size="circle"
+            onClick={onWarmTransferClick}
+          >
+            {TaskHelper.isChatBasedTask(task) ? (
+              <ChatIcon key={`queue-item-warm-transfer-icon-${queue.sid}`} decorative={false} title="Warm Transfer" />
+            ) : (
+              <CallTransferIcon
+                key={`queue-item-warm-transfer-icon-${queue.sid}`}
+                decorative={false}
+                title="Warm Transfer"
+              />
+            )}
+          </Button>
+        </Tooltip>
+        <Tooltip
+          key={`queue-item-buttons-cold-transfer-tooltip-${queue.sid}`}
+          element="TRANSFER_DIR_COMMON_TOOLTIP"
+          text="Cold Transfer"
+        >
+          <Button
+            key={`queue-item-warm-transfer-button-${queue.sid}`}
+            variant="secondary_icon"
+            size="circle"
+            onClick={onColdTransferClick}
+          >
+            {TaskHelper.isChatBasedTask(task) ? (
+              <SendIcon key={`queue-item-cold-transfer-icon-${queue.sid}`} decorative={false} title="Warm Transfer" />
+            ) : (
+              <CallOutgoingIcon
+                key={`queue-item-cold-transfer-icon-${queue.sid}`}
+                decorative={false}
+                title="Warm Transfer"
+              />
+            )}
+          </Button>
+        </Tooltip>
+      </ButtonGroup>
```

Comment:
There are a few scenarios where the warm transfer icon should not render:
- The warm transfer feature has been disabled (or was never enabled in the first place) in Flex Admin
- Conversation transfer has a config option for warm transfer. If that option is disabled, the warm transfer icon should be hidden for CBM tasks.

For the conversation-transfer feature item, [this code](https://github.com/twilio-professional-services/flex-project-template/blob/main/plugin-flex-ts-template-v2/src/feature-library/conversation-transfer/flex-hooks/actions/ShowDirectory.ts) is responsible for hiding the button. Best case scenario is we are able to use css class names such that the existing hiding code works for this as well, but if that isn't possible then checking the config might be needed.

I see a `isWarmTransferEnabled` prop getting passed down here but it is unused. It should probably be used here and then set higher up depending on the above items.

Reply:
It looks like the `isWarmTransferEnabled` prop out-of-the-box is set by the warm transfer feature flag, which gets you half way here :)